File: problems/quest/data_structures_and_algorithms/arrays/Max_Consecutive_Ones.py
from typing import List


class MaxConsecutiveOnes:
    def find_max_consecutive_ones(self, nums: List[int]) -> int:
        max_streak = 0
        cur_streak = 0

        for num in nums:
            if num == 1:
                cur_streak += 1
                if cur_streak > max_streak:
                    max_streak = cur_streak
            else:
                cur_streak = 0

        return max_streak

File: problems/quest/data_structures_and_algorithms/arrays/MaxConsecutiveOnes.java
   class MaxConsecutiveOnes {
    // за каждый проход по циклу выполняется проверка - не очень
    public int findMaxConsecutiveOnes1(int[] nums) {
        int max_consecution = 0;
        int cur_max_consecution = 0;

        for (int num : nums) {
            if (num == 0) cur_max_consecution = 0;
            else cur_max_consecution += 1;
            
            if (cur_max_consecution > max_consecution)
                max_consecution = cur_max_consecution;
        }
        return max_consecution;
    }

    // мое более оптимизированное решение
    public int findMaxConsecutiveOnes2(int[] nums) {
        int max_consecution = 0;
        int cur_max_consecution = 0;

        for (int num : nums) {
            if (num == 1) {
                cur_max_consecution++;
                
                if (cur_max_consecution > max_consecution)
                max_consecution = cur_max_consecution;
            }
            else cur_max_consecution = 0;
        }
        return max_consecution;
    }
}
